common-app-lib/src/cmr/common_app/services/search/query_validation.clj

```diff
@@ -24,9 +25,11 @@
                         result-format)]
     (when-not (get (supported-result-formats concept-type) result-format)
       (if-let [version (mt/version-of mime-type)]
-        [(format "The mime type [%s] with version [%s] is not supported for %ss."
-                 (mt/base-mime-type-of mime-type) version (name concept-type))]
-        [(format "The mime type [%s] is not supported for %ss." mime-type (name concept-type))]))))
+        ;; Some concepts aren't plural with just an 's'
+        [(format "The mime type [%s] with version [%s] is not supported for %s."
+                 (mt/base-mime-type-of mime-type) version (inf/plural (name concept-type)))]
```

Comment:
That is fair, I would maybe push back that it is already being used in the Common App project: `[inflections "0.13.0"]` I didn't really introduce it, I can look for an alternative way to pluralize though.

Reply:
What I would suggest in common-lib/src/cmr/common/concepts.clj on line 80 there is a function called pluralize-concept-type which takes a singular concept-key (such as :collection) and returns the plural key (:collections).  Since you need the name and not the key, could you please add 2 new functions into concept.clj

(defn pluralize-concept-type-name
  "Pluralizes the passed in concept keyword/string and returns the string. The compliment
   function is singularize-concept-type-name"
  [concept-key]
  (inf/plural (name concept-key)))

(defn singularize-concept-type-name
  "Singularize the passed in concept keyword/string and returns the string. The compliment
   function is pluralize-concept-type-name"
  [concept-key]
  (inf/singular (name concept-key)))

Then change starting on line 80 (the old number) the function to be:
(defn pluralize-concept-type
  "Pluralizes the passed in concept keyword/string and returns it. The compliment
   function is singularize-concept-type"
  [concept-key]
  (keyword (pluralize-concept-type-name concept-key)))

(defn singularize-concept-type
  "Singularize the passed in concept keyword/string and returns it. The compliment
   function is pluralize-concept-type"
  [concept-key]
  (keyword (singularize-concept-type-name concept-key)))

Then in your code use the pluralize-concept-type-name function.  I agree with Lucia that should use the wrapper when we can.  

Doing a search through the code, I do see it being used in many places where we should use the wrapper.  I will write a ticket to change these to use the common-lib functions.

Reply:
Added the functions and swapped according to @eereiter comment